cli/tools/installer.rs

```diff
@@ -398,7 +400,21 @@ fn resolve_shim_data(
     ));
   }
 
-  if flags.no_lock {
+  let config = CliOptions::from_flags(flags.clone())?;
+
+  let no_lock = if let Some(no_lock) = flags.no_lock {
+    no_lock
+  } else if let Some(lock_config) = config.to_lock_config()? {
+    if let LockConfig::Bool(lock) = lock_config {
+      !lock
+    } else {
+      true
+    }
+  } else {
+    false
+  };
```

Comment:
Since `deno install` is more of a global command, the install shouldn't consult the configuration. So this should still just be `if flags.no_lock {` below.

Reply:
Done!